management/access_control/src/access_control.cc

```diff
@@ -91,102 +99,105 @@ std::string AccessControl::GenerateCookie() {
-    failed_cmd_pub_.publish(cmd);
+    failed_cmd_pub_->publish(cmd);
   }
 }
 
-void AccessControl::HandleGrabControl(ff_msgs::CommandStampedConstPtr const&
-                                                                          cmd) {
+void AccessControl::HandleGrabControl(ff_msgs::CommandStamped const& cmd) {
   std::string msg;
-  if (cmd->args.size() != 1) {
+  if (cmd.args.size() != 1) {
     msg = "Invalid number of arguments to grab control command.";
-    ROS_WARN("%s", msg.c_str());
-    PublishAck(cmd->cmd_id,
+    FF_WARN("%s", msg.c_str());
+    PublishAck(cmd.cmd_id,
                msg,
                ff_msgs::AckCompletedStatus::BAD_SYNTAX);
     return;
   }
 
-  if (cmd->args[0].data_type != ff_msgs::CommandArg::DATA_TYPE_STRING) {
+  if (cmd.args[0].data_type != ff_msgs::CommandArg::DATA_TYPE_STRING) {
     msg = "Invalid argument type to grab control command.";
-    ROS_WARN("%s", msg.c_str());
-    PublishAck(cmd->cmd_id,
+    FF_WARN("%s", msg.c_str());
+    PublishAck(cmd.cmd_id,
                msg,
                ff_msgs::AckCompletedStatus::BAD_SYNTAX);
     return;
   }
 
-  const std::string cookie = cmd->args[0].s;
+  const std::string cookie = cmd.args[0].s;
   if (cookie != state_.cookie) {
-    msg = "Agent " + cmd->cmd_src + " sent wrong cookie. Expected "
+    msg = "Agent " + cmd.cmd_src + " sent wrong cookie. Expected "
           + state_.cookie + ", got " + cookie;
-    ROS_WARN("%s", msg.c_str());
-    PublishAck(cmd->cmd_id,
+    FF_WARN("%s", msg.c_str());
+    PublishAck(cmd.cmd_id,
                msg,
                ff_msgs::AckCompletedStatus::EXEC_FAILED);
     return;
   }
 
-  if (cmd->cmd_src != requestor_) {
-    ROS_WARN("Agent %s has grabbed control, yet %s requested it",
-             cmd->cmd_src.data(), requestor_.data());
+  if (cmd.cmd_src != requestor_) {
+    FF_WARN("Agent %s has grabbed control, yet %s requested it",
+             cmd.cmd_src.data(), requestor_.data());
   }
 
-  state_.controller = cmd->cmd_src;
+  state_.controller = cmd.cmd_src;
   state_.cookie = "";
 
   PublishState();
-  PublishAck(cmd->cmd_id);
+  PublishAck(cmd.cmd_id);
   return;
 }
 
-void AccessControl::HandleRequestControl(ff_msgs::CommandStampedConstPtr const&
-                                                                          cmd) {
+void AccessControl::HandleRequestControl(
+                                      ff_msgs::CommandStamped const& cmd) {
   state_.cookie = GenerateCookie();
-  requestor_ = cmd->cmd_src;
+  requestor_ = cmd.cmd_src;
 
-  PublishAck(cmd->cmd_id);
+  PublishAck(cmd.cmd_id);
   PublishState();
 }
 
 void AccessControl::PublishAck(std::string const& cmd_id,
                                std::string const& message,
                                uint8_t completed_status,
                                uint8_t status) {
-  ack_.header.stamp = ros::Time::now();
+  ack_.header.stamp = node_->get_clock()->now();
```

Comment:
Use the function you made?

Reply:
Good catch. Marina actually made that function.